packages/dma-contracts/contracts/core/OperationsRegistry.sol

```diff
@@ -29,41 +29,45 @@ contract OperationsRegistry {
   }
 
   /**
-   * @notice Stores the Actions that constitute a given Operation
-   * @param newOwner The address of the new owner of the Operations Registry
-   */
-  function transferOwnership(address newOwner) public onlyOwner {
-    owner = newOwner;
-  }
-
-  /**
-   * @dev Emitted when a new operation is added or an existing operation is updated
+   * @dev Emitted when a new operation is added.
    * @param name The Operation name
+   * @param operationHash A hashed value of all action hashes ( concatenated )
    **/
-  event OperationAdded(bytes32 indexed name);
+  event OperationAdded(string name, bytes32 operationHash);
 
   /**
-   * @notice Adds an Operation's Actions keyed to a an operation name
-   * @param operation Struct with Operation name, actions and their optionality
+   * @notice Adds an operation under specific name for a specific ordered list of actions
+   * @dev Each action is stored in a ServiceRepository under a specific hash.
+   * To get the operationHash, one should concatenated all hashes of the actions
+   * that are used in the specific operation and hash them using keccak256.
+   * The order of the actions matters since different order will result in different hash value
+   * hence difference operation.
+   * @param name Name of the operation that is going to be stored
+   * @param operationHash A hash of all action hashes ( concatenated )
    */
-  function addOperation(StoredOperation calldata operation) external onlyOwner {
-    operations[operation.name] = operation;
+  function addOperation(string memory name, bytes32 operationHash) external onlyOwner {
+    operations[operationHash] = bytes32(bytes(name));
     // By packing the string into bytes32 which means the max char length is capped at 64
-    emit OperationAdded(bytes32(abi.encodePacked(operation.name)));
+    emit OperationAdded(name, operationHash);
   }
 
   /**
-   * @notice Gets an Operation from the Registry
-   * @param name The name of the Operation
-   * @return actions Returns an array of Actions and array for optionality of coresponding Actions
+   * @notice Returns either a valid name of an added operation or reverts with an error if no operation is found.
+   * @param operationHash A hashed value of all action hashes ( concatenated ).
    */
-  function getOperation(
-    string memory name
-  ) external view returns (bytes32[] memory actions, bool[] memory optional) {
-    if (keccak256(bytes(operations[name].name)) == keccak256(bytes(""))) {
-      revert("Operation doesn't exist");
+  function getOperationName(bytes32 operationHash) external view returns (bytes32 name) {
```

Comment:
It should return `(string name)` so it uses the same interface as `addOperation`. Otherwise I'd change `addOperation` to use `bytes32 name` as well